proposals/security_and_privacy.md

```diff
@@ -0,0 +1,277 @@
+
+### Does this specification introduce new state for an origin that persists across browsing sessions? 
+
+This is not a browser-oriented specification although
+WoT Things can be accessed through browsers as if they
+were web services. 
+However, no new state is introduced or required in browsers beyond
+current mechanisms (cookies, etc).
+
+WoT Things can also act as clients, however, accessing
+web services and other WoT Things acting as servers.
+It is also possible that the same device could instantiate
+multiple WoT Runtimes.  However, there is noi explicit mechanism
+for sharing state between such instances, and each instance
+can be considered a single origin.
+
+### Does this specification expose persistent, cross-origin state to the web? 
+
+A WoT Thing may act as a server and expose interactions that 
+allow properties to be modified.  Therefore modifications by
+one client may be visible to another.
+
+This risk of this, if any, is mitigated by the fact that
+WoT Things support various mechanisms to limit access to 
+authorized users, like any other web service.
+
+When acting as a client, each WoT Thing is in effect single-origin.
+
+There is a potential fingerprinting risk, which is discussed,
+and is related to the PII risk discussed above.
+This can be mitigated by controlling access to WoT metadata
+such as WoT Thing Descriptions.
+
+### Does this specification expose any other data to an origin that it doesn’t currently have access to?
+
+Yes.
+
+WoT Things by nature will be connected into a network of
+devices.  A single "WoT Network" application may involve
+communications between many WoT Things, some acting as
+servers, some acting as clients, some acting as both.
+
+In addition, WoT Things may act as gateways to other
+non-HTTP devices using other network protocols, such
+as MQTT or CoAP, or to interface to a variety of non-IP
+devices using BACNET, ECHONET, X10, SPI, I2C, and many
+other possibilities.
+
+Unfortunately, the single-origin model used in 
+web servers and browsers does not scale well to this model.
+
+In practice, securing an IoT Application in general is
+out of scope.  We do however discuss several common connection
+patterns (client-server, proxy, etc) and how they can be secured.
+
+This risk is mitigated by the fact that endpoint IoT devices
+are generally not exposed directly on the internet, but
+are mediated by gateways and cloud services.
+
+### Does this specification enable new script execution/loading mechanisms? 
+
+Yes.  It enables WoT Things to run applications based
+on ECMAScript, using a constrained API, and with suitable
+restrictions, such as no direct access to private
+security data.
+
+However, a mechanism for loading new scripts is not defined
+and is currently considered to be out of scope.
+Our architecture assumes the scripts running inside a WoT Thing
+are trusted implementations provided by the manufacturer
+of the device and installed using a suitable OTA update
```

Comment:
I would put OTA in brackets, since you could have other mechanisms too

Reply:
True, it was just meant to be an example.   Maybe I should just say "for example...".   BTW I plan to update the security considerations section of the main document to say something similar.  It occurs to me we did not discuss scope, operational phase, etc. in the main document.   I probably should also talk about cross-site issues, single origin, shared data, Zero-Trust, etc.  Most of these are irrelevant to IoT, or rather, a different approach needs to be taken to software distribution.   This is why I was reluctant to include scripts in the TD.  If we did include them avoiding cross-site scripting hacks would be a real nightmare given the more general connections used in IoT architectures.

Reply:
I decided to just remove OTA, it's not really relevant whether the update is OTA or not.